### PR TITLE
feat(core-distributor)!: add cuid cache

### PR DESCRIPTION
**Pull request by gurinderu**
## Description
Allows to cache assignment by cuid for the core distributor.

## Motivation
physical_core_id stability is critical for the CCP 

## Proposed Changes
Added cache for cuid->physical_core_id

## Checklist
- [ ] The code follows the project's coding conventions and style guidelines.
- [ ] All tests related to the changes have passed successfully.
- [ ] Documentation has been updated to reflect the changes (if applicable).
- [ ] All new and existing unit tests have passed.
- [ ] I have self-reviewed my code and ensured its quality.
- [ ] I have added/updated necessary comments to aid understanding.

## Reviewer Checklist
- [ ] Code has been reviewed for quality and adherence to [guidelines](https://doc.rust-lang.org/1.0.0/style/README.html).
- [ ] Tests have been reviewed and are sufficient to validate the changes.
- [ ] Documentation has been reviewed and is up to date.
- [ ] Any questions or concerns have been addressed.

### CODE REVIEW

File: crates/core-distributor/src/persistence.rs [hunk 1]
```diff
@@ -95,6 +96,8 @@ pub struct PersistentCoreDistributorState {
     pub unit_id_mapping: Vec<(PhysicalCoreId, CUID)>,
     #[serde_as(as = "Vec<(Hex, _)>")]
     pub work_type_mapping: Vec<(CUID, WorkType)>,
+    #[serde_as(as = "Vec<(Hex, _)>")]
+    pub cpu_cache: Map<CUID, PhysicalCoreId>,
```
Comment (folex) [inline crates/core-distributor/src/persistence.rs:100]: Are you sure you need to persist a cache to disk?

Comment (gurinderu) [reply crates/core-distributor/src/persistence.rs]: I'm not sure, but it can be better in case of restarts.

Comment (mikevoronov) [reply crates/core-distributor/src/persistence.rs]: Am I right that providers will need to cleanup cache to update Nox in this case?

Comment (gurinderu) [reply crates/core-distributor/src/persistence.rs]: Yes, we need to cleanup at least the core distributor state

**Review by mikevoronov — APPROVED** [review]
utACK

**Review by justprosh — APPROVED** [review]

**Pull request merged**